### ISSUE TITLE
t/date-parse.t started to fail in 2020

### ISSUE THREAD

**Issue by eserte** [barefootcoder/date-easy#14]
See subject. Excerpt of test log:
```
#   Failed test 'successful 0 parse: 3155760000'
#   at t/date-parse.t line 123.
#          got: '2070-01-01'
#     expected: '1970-01-01'
# Looks like you failed 1 test of 733.
t/date-parse.t ........... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/733 subtests 
```
http://matrix.cpantesters.org/?dist=Date-Easy%200.09;reports=1#sl=7,1 also shows that reports generated since 2020-01-01 are fail reports.

**Comment by barefootcoder** [comment]
This is remarkably similar to #12, where a bug in `str2time` caused a test failure (which did not indicate a failure in the module).  This time, I'm bitten by `timelocal`'s stupid "rolling century" because now the beginning of the Unix epoch is 50 years ago.  So I'm trying to convert `0` to the local time via `timelocal gmtime 0`, and stupid `gmtime` returns `70` for the year, and stupid `timelocal` (now) thinks that means 2070 instead of 1970.  My goto fix for these sorts of problems (dropping in `timelocal_modern`) doesn't fix it, because _that_ just makes it think I want the year 70AD. :-/

I'm going to need to ponder if I even really need this test and, if so, how best to fix it.  Thx as always for the prompt report.  (And, remember: it's not a bug in the module, just the test.)

**Comment by eserte** [comment]
Unfortunately timelocal/timegm are not the inverse functions to localtime/gmtime. You have to add 1970 to the year part returned by localtime/gmtime.

**Comment by barefootcoder** [comment]
> Unfortunately timelocal/timegm are not the inverse functions to localtime/gmtime.

Indeed.  I just found (or perhaps refound) [this article](http://blogs.perl.org/users/grinnz/2019/07/the-timelocal-trap.html), which explains it pretty well.

I'm not entirely sure I need to keep this test anyway, but I think I'll fix it as an exercise.  I want to do one other feature add while I'm in there, but hopefully I'll have something in the next day or so.

And perhaps I can squeeze a blog post out of it as well. :-)

**Comment by barefootcoder** [comment]
(Sorry for the delay; my laptop completely melted down on Wednesday and I'm just now back to a semblance of normality.)

Okay, fix is uploaded as 0.09_01; we'll see what CPAN Testers has to say.  Working on the blog post now. :-)

**Comment by barefootcoder** [comment]
Okay, [0.10](https://metacpan.org/release/Date-Easy) is available on CPAN now.  This issue should be resolved.
